src/View/AbstractDirectoryView.vala

```diff
@@ -827,7 +827,12 @@ namespace FM {
                     }
                 }
             } else {
-                warning ("Cannot open file in trash");
+                PF.Dialogs.show_error_dialog (
+                    ///TRANSLATORS: '%s' is a quoted placehorder for the name of a file. It can be moved but not omitted
+                    _("The file “%s” cannot be opened because it is in the Trash").printf (file.basename),
+                    _("To use this file, first drag it out of the Trash"),
```

Comment:
This copy should probably say something more about "restore" the item instead of being too specific with a drag operation since there are multiple ways to restore a trashed file

Reply:
Strictly speaking you do not have to "restore" the file to its original location - just move it out of the trash.